client.go

```diff
@@ -304,13 +305,16 @@ func SplitVersionedURL(url string) (string, string, bool) {
 // BaseURL should refer to the root of the MAAS server path, e.g.
 // http://my.maas.server.example.com/MAAS/
 // apiVersion should contain the version of the MAAS API that you want to use.
-func NewAnonymousClient(BaseURL string, apiVersion string) (*Client, error) {
+func NewAnonymousClient(BaseURL string, apiVersion string, insecure bool) (*Client, error) {
```

Comment:
Technically we would need a v3 for this, as it becomes incompatible with existing v2 usages.

Have you considered using the [variadic options pattern(https://www.sohamkamani.com/golang/options-pattern/)? This would mean existing constructor usage in the wild would remain compatible.

Reply:
Thank you for the hint. Based on that, I decided to implement a more generic function that sets TLS configuration in general and not only insecureSkipVerify. In addition, no change is required to `gomaasapi` since you are exposing the HTTPClient already so I can set it from the calling function. As such, this PR is irrelevant.